### DISCUSSION TITLE
Redirect before render

### DISCUSSION THREAD

**Discussion by labrocadabro** [the-collab-lab/tcl-60-smart-shopping-list#23] [General]
@paramsiddharth, @RajGM, @iamkrati22: This is something from our issue a couple of weeks ago that's been bothering me & I wanted to get some feedback. 

When a user visits the homepage with a token set, it redirects them to the list page. However, because the redirection happens within useEffect, the home component renders first, then the token is checked and the user is redirected. This leads to a flash of the homepage showing before the redirect. 

https://user-images.githubusercontent.com/5794319/233901106-21a4d490-ddf4-4b4b-b03b-065a29e9ee4b.mov

Is there any way to redirect before rendering so the homepage doesn't flash? It's a minor issue but I'm curious if there's a good way to deal with it.

**Comment by RajGM** [comment]
Hi @labrocadabro 
We can use conditional rendering for verifying token and based on that we can decide to redirect or not.

**Comment by paramsiddharth** [comment]
Interesting point! We'll discuss this in today's weekly sync and I'll share my perspective of what should solve this.

**Comment by labrocadabro** [comment]
Thank you! This may be addressed by your suggestions, but I noticed today that in a particular situation, the redirect is 'trapping' the back button. 

When you start from the home page, then create a new list or enter a token, you're redirected to the /list page. Then if you try to hit the back button, you're immediately pushed back to the /list page, because the previous page you were on was the home page, but that is set to automatically redirect you back to the /list page. It doesn't happen if you visit the homepage with a token already set, I assume because you are immediately redirected to the /list page and you never actually visit the homepage. 

That makes this is an edge case situation, and not one many people would encounter - but I was wondering, is there a way to remove the homepage from the history (or to put it another way, replace the homepage with the /list page) when the redirect happens?

I did take a look at the react router docs but they don't seem to have search (!) and I couldn't find anything that answered this question.
